### PR TITLE
Set Kibana public domain based on Authentication domain

### PR DESCRIPTION
**Pull request by tmjd**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/logstorage.go [hunk 1]
```diff
@@ -207,6 +207,7 @@ type ElasticsearchConfiguration struct {
 	KbService                   *corev1.Service
 	ClusterDomain               string
 	DexCfg                      DexRelyingPartyConfig
+	ManagerDomain               string // Domain where the manager is reachable, for setting Kibana publicBaseUrl
```
Comment (rene-dekker) [inline pkg/render/logstorage.go:210]: If this variable would be called KibanaPublicURL, you would need no comment to clarify it and `fmt.Sprintf("%s/%s", es.cfg.ManagerDomain, KibanaBasePath)`  could then be done in the controller, saving you an if-statement on line 1263. WDYT?

**Review by rene-dekker — APPROVED** [review]
lgtm

File: pkg/render/logstorage.go (hunk 1 above)
Comment (tmjd) [reply pkg/render/logstorage.go:210]: I like the idea but I think it is more appropriate to construct it in the render, that is the correct scope for doing that.

File: pkg/render/logstorage.go [hunk 2]
```diff
@@ -1259,8 +1260,15 @@ func (es elasticsearchComponent) kibanaCR() *kbv1.Kibana {
 		"defaultRoute":    fmt.Sprintf(KibanaDefaultRoute, TimeFilter, url.PathEscape(FlowsDashboardName)),
 	}
 
-	if es.cfg.DexCfg != nil && es.cfg.DexCfg.BaseURL() != "" {
-		server["publicBaseUrl"] = fmt.Sprintf("%s/tigera-kibana", es.cfg.DexCfg.BaseURL())
+	baseUrl := es.cfg.ManagerDomain
+	// If we can't parse the ManagerDomain with url.Parse then assume it won't be valid for the Kibana publicBaseUrl
+	if u, err := url.Parse(baseUrl); err == nil {
```
Comment (Brian-McM) [inline pkg/render/logstorage.go:1265]: It might be a good idea to log the error, might be hard to debug if we don't.

**Review by Brian-McM — APPROVED** [review]

**Pull request merged**